File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/mpim/MpimOpenRequest.java
package com.github.seratch.jslack.api.methods.request.mpim;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

import java.util.List;

public class MpimOpenRequest implements SlackApiRequest {

    /**
     * Authentication token. Requires scope: `mpim:write`
     */
    private String token;

    /**
     * Comma separated lists of users.
     * The ordering of the users is preserved whenever a MPIM group is returned.
     */
    private List<String> users;

    MpimOpenRequest(String token, List<String> users) {
        this.token = token;
        this.users = users;
    }

    public static MpimOpenRequestBuilder builder() {
        return new MpimOpenRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public List<String> getUsers() {
        return this.users;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public void setUsers(List<String> users) {
        this.users = users;
    }

    public static class MpimOpenRequestBuilder {
        private String token;
        private List<String> users;

        MpimOpenRequestBuilder() {
        }

        public MpimOpenRequest.MpimOpenRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public MpimOpenRequest.MpimOpenRequestBuilder users(List<String> users) {
            this.users = users;
            return this;
        }

        public MpimOpenRequest build() {
            return new MpimOpenRequest(token, users);
        }

        public String toString() {
            return "MpimOpenRequest.MpimOpenRequestBuilder(token=" + this.token + ", users=" + this.users + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/mpim/MpimOpenRequest.kt
package work.socialhub.kslack.api.methods.request.mpim

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest
import kotlin.js.JsExport

@JsExport
class MpimOpenRequest(
    /** Authentication token. Requires scope: `mpim:write` */
    override var token: String?,

    /**
     * Comma separated lists of users.
     * The ordering of the users is preserved whenever a MPIM group is returned.
     */
    var users: Array<String>?
) : SlackApiRequest, FormRequest {

    override fun toMap(): Map<String, Any> {
        return mutableMapOf<String, Any>().also {
            if (users != null) {
                it.addParam("users", users!!.joinToString(","))
            }
        }
    }
}